### PR TITLE
Add automation tests for upgrade aws workload cluster

### LINKED ISSUE #2457: TC08: AWS Cluster - Able to upgrade CC workload cluster on AWS and verify if all addons are successfully upgraded
Since the latest version is TKr 1.23, I'll try to upgrade from 1.22 to 1.23.

### PR DESCRIPTION
**Pull request by maralavi**
Signed-off-by: Marjan Alavi <malavi@vmware.com>

### What this PR does / why we need it
Add automation tests for upgrade aws workload cluster

### Which issue(s) this PR fixes

Fixes #2457

### Release note

```release-note
Add automation tests for upgrade aws workload cluster
```

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/aws_cc/aws_cc_suite_test.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 
 	By(fmt.Sprintf("Loading the e2e test configuration from %q", e2eConfigPath))
 	e2eConfig = framework.LoadE2EConfig(context.TODO(), framework.E2EConfigInput{ConfigPath: e2eConfigPath})
-	Expect(e2eConfigPath).ToNot(BeNil(), "Failed to load e2e config from %s", e2eConfigPath)
+	Expect(e2eConfig).ToNot(BeNil(), "Failed to load e2e config from %s", e2eConfigPath)
```
Comment (adduarte) [inline pkg/v1/tkg/test/tkgctl/aws_cc/aws_cc_suite_test.go:74]: why the change? unless it changes code behavior, even if it is really ugly name don't change variable names unless you have to.

File: pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go [hunk 2]
```diff
@@ -116,6 +124,32 @@ func E2ETKRResolverValidationForClusterCRUDSpec(context context.Context, inputGe
 		clusterConfigFile, err = framework.GetTempClusterConfigFile(input.E2EConfig.TkgClusterConfigPath, &options)
 		Expect(err).To(BeNil())
 
+		if input.OtherConfigs != nil {
```
Comment (adduarte) [inline pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go:127]: I don't think you need this

the format 
if isClusterClass, ok := input.OtherConfigs["clusterclass"];

Is meant just for that. If input.OtherConfigs is nil , ok evaluates to false... 
check this out, you can run this: 
```
type spec struct {
	OtherConfigs map[string]string
}
func main() {
	input := spec{}
	if isClusterClass, ok := input.OtherConfigs["clusterclass"]; ok && isClusterClass == "true" {
		println("what???? this is not possible")
	}
	if input.OtherConfigs == nil {
		println("This works ")
	}
}

File: pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go [hunk 3]
```diff
@@ -126,6 +160,16 @@ func E2ETKRResolverValidationForClusterCRUDSpec(context context.Context, inputGe
 			SkipPrompt:  true,
 		})
 		Expect(err).To(BeNil())
+
+		// verify addon resources are deleted successfully in clusterclass mode
+		if input.OtherConfigs != nil {
```
Comment (adduarte) [inline pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go:165]: same as before, but maybe something made you put it here I don't see

**Review by adduarte — COMMENTED** [review]
Looks good. Nothing I would require changes to.

File: pkg/v1/tkg/test/tkgctl/aws_cc/aws_cc_suite_test.go (hunk 1 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/aws_cc/aws_cc_suite_test.go:74]: Cause it was a bug before. ```e2eConfig``` has just been created in the line above should have been tested for nilness not ```e2eConfigPath```

Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/aws_cc/aws_cc_suite_test.go:74]: I haven't changed variable names

Comment (adduarte) [reply pkg/v1/tkg/test/tkgctl/aws_cc/aws_cc_suite_test.go:74]: never mind. I see the reason.

File: pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go (hunk 2 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go:127]: True, I can remove that check. Will do, so as to get rid of one unnecessary indentation level too.

File: pkg/v1/tkg/test/tkgctl/shared/upgrade.go [hunk 4]
```diff
@@ -175,11 +175,11 @@ func E2EUpgradeSpec(context context.Context, inputGetter func() E2EUpgradeSpecIn
 	})
 }
 
-func validateKubernetesVersion(clusterName string, expectedK8sVersion string) { // nolint:unused
+func validateKubernetesVersion(clusterName, expectedK8sVersion, kubeconfigPath string) { // nolint:unused
```
Comment (HL-EverGreen) [inline pkg/v1/tkg/test/tkgctl/shared/upgrade.go:178]: question: why do we need to pass in kubeconfig?

File: pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go [hunk 5]
```diff
@@ -191,16 +257,30 @@ func E2ETKRResolverValidationForClusterCRUDSpec(context context.Context, inputGe
...
 		By(fmt.Sprintf("Validating the kubernetes version after cluster %q is upgraded", clusterName))
-		validateKubernetesVersion(clusterName, defaultTKR.Spec.Kubernetes.Version)
+		validateKubernetesVersion(clusterName, defaultTKR.Spec.Kubernetes.Version, tempKubeConfigFilePath)
 
 		By(fmt.Sprintf("Validating the TKR data after cluster %q is upgraded", clusterName))
 		verifyTKRData(context, mcProxy, options.ClusterName, options.Namespace)
 
+		// verify addons are deployed successfully in clusterclass mode
+		if isClusterClass, ok := input.OtherConfigs["clusterclass"]; ok && isClusterClass == "true" {
+			By(fmt.Sprintf("Verify addon packages on management cluster %q matches clusterBootstrap info on management cluster %q", input.E2EConfig.ManagementClusterName, input.E2EConfig.ManagementClusterName))
```
Comment (HL-EverGreen) [inline pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go:271]: nit: maybe considering add word like "after cluster upgrade" to differentiate with the check before the upgrade.

**Review by haoranleo — APPROVED** [review]
Looks good to me!

File: pkg/v1/tkg/test/tkgctl/shared/upgrade.go (hunk 4 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/upgrade.go:178]: Cause ```framework.NewClusterProxy()``` uses that and it was passing "" which uses the default kubeconfig and wouldn't work for the kubeconfig created in the temp directory

File: pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go (hunk 5 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go:271]: Could I record this and make this change in a subsequent PR? the pipeline takes a long time to run.

**Review by anujc25 — APPROVED** [review]
